src/PSR7/Validators/BodyValidator/MultipartValidation.php

```diff
@@ -0,0 +1,187 @@
+     *
+     * @param MediaType[] $mediaTypeSpecs
+     */
+    private function validateMultipart(OperationAddress $addr, MessageInterface $message, array $mediaTypeSpecs, string $contentType) : void
+    {
+        /** @var Schema $schema */
+        $schema = $mediaTypeSpecs[$contentType]->schema;
+
+        // 0. Multipart body message MUST be described with a set of object properties
+        if ($schema->type !== CebeType::OBJECT) {
+            throw TypeMismatch::becauseTypeDoesNotMatch('object', $schema->type);
+        }
+
+        // 1. Parse message body
+        $document = PSR7::convert($message);
+
+        // 2. Validate bodies of each part
+        $body = $this->parseMultipartData($addr, $document);
+
+        $validator = new SchemaValidator($this->detectValidationStrategy($message));
+        try {
+            $validator->validate($body, $schema);
+        } catch (SchemaMismatch $e) {
+            throw InvalidBody::becauseBodyDoesNotMatchSchema($contentType, $addr, $e);
+        }
+
+        // 3. Validate specified part encodings and headers
+        // @see https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#encoding-object
+        // The encoding object SHALL only apply to requestBody objects when the media type is multipart or application/x-www-form-urlencoded.
+        // An encoding attribute is introduced to give you control over the serialization of parts of multipart request bodies.
+        // This attribute is only applicable to "multipart" and "application/x-www-form-urlencoded" request bodies.
+        $encodings = $mediaTypeSpecs[$contentType]->encoding;
+
+        foreach ($encodings as $partName => $encoding) {
+            $parts = $document->getPartsByName($partName); // multiple parts share a name?
+            if (! $parts) {
+                throw new RuntimeException(sprintf(
+                    'Specified body part %s is not found',
+                    $partName
+                ));
+            }
+
+            foreach ($parts as $part) {
+                // 3.1 parts encoding
+                $partContentType     = $part->getHeader(self::HEADER_CONTENT_TYPE);
+                $encodingContentType = $this->detectEncondingContentType($encoding, $part, $schema->properties[$partName]);
+                if (strpos($encodingContentType, '*') === false) {
+                    // strict comparison (ie "image/jpeg")
+                    if ($encodingContentType !== $partContentType) {
+                        throw InvalidBody::becauseBodyDoesNotMatchSchemaMultipart(
+                            $partName,
+                            $partContentType,
+                            $addr
+                        );
+                    }
+                } else {
+                    // loose comparison (ie "image/*")
+                    $encodingContentType = str_replace('*', '.*', $encodingContentType);
```

Comment:
i'm not sure this is a proper way. there are multiple valid content-types like `application/xhtml+xml` which whould render as invalid (or unexpectedly valid) regular expression here

Reply:
What would be a better approach to this validation? Escaping symbols manually?

Reply:
We can [`preg-quote`](use https://www.php.net/manual/en/function.preg-quote.php) here and then manually unescape `*`, i think